components/script/script_thread.rs

```diff
@@ -1888,6 +1894,18 @@ impl ScriptThread {
         warn!("resize sent to nonexistent pipeline");
     }
 
+    fn handle_exit_fullscreen(&self, id: PipelineId) {
+        let document = self.documents.borrow().find_document(id);
+        if let Some(document) = document {
+            let _ac = JSAutoCompartment::new(
```

Comment:
Add this comment: `// exit_fullscreen creates a new JS promise object, so we need to have entered a compartment.`